File: ruby/exercism/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  attr_reader :colors

  COLOR_MATCHES = {
    "black" => 0,
    "brown" => 1,
    "red" => 2,
    "orange" => 3,
    "yellow" => 4,
    "green" => 5,
    "blue" => 6,
    "violet" => 7,
    "grey" => 8,
    "white" => 9
  }

  def initialize(colors)
    @colors = colors
  end

  def self.value(colors)
    new(colors).value
  end

  def value
    numbers.join.to_i
  end

  private

  def numbers
    colors.map do |color|
      number = COLOR_MATCHES[color.downcase]
      raise ArgumentError if number.nil?
      number
    end
  end
end

File: js/exercism/resistor-color-duo/resistor-color-duo.js
//
// This is only a SKELETON file for the 'Resistor Color Duo' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

const match = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];

const colorCode = color => match.indexOf(color);

export const value = ([c1, c2]) => {
  return colorCode(c1)*10 + colorCode(c2);
};
